File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 2 - home/DelayTaskRunner.java
public class DelayTaskRunner extends AbstractTaskRunner {
    public DelayTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        try {
            System.out.println("Waiting for task to complete...");
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        super.executeOneTask();
    }
}

File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 11 - homework/lab11/lab11/task/DelayTaskRunner.cs
using System.Security.AccessControl;

namespace lab11;

public class DelayTaskRunner:AbstractTaskRunner
{
    public DelayTaskRunner(TaskRunner taskRunner) : base(taskRunner)
    {
    }

    public override void ExecuteOneTask()
    { 
        Thread.Sleep(3000);
        base.ExecuteOneTask();
        
        
    }

    public override void ExecuteAll()
    {
        Thread.Sleep(3000);
        base.ExecuteAll();
    }

    public override void AddTask(Task task)
    {
        Thread.Sleep(3000);
        base.AddTask(task);
    }

    public override bool HasTasks()
    {
        Thread.Sleep(3000);
        return base.HasTasks();
    }
}